webapp/channels/src/components/channel_header_menu/menu_items/channel_settings_submenu.tsx

```diff
@@ -0,0 +1,157 @@
+    CogOutlineIcon,
+} from '@mattermost/compass-icons/components';
+import type {Channel} from '@mattermost/types/channels';
+
+import {Permissions} from 'mattermost-redux/constants';
+
+import {openModal} from 'actions/views/modals';
+
+import ConvertChannelModal from 'components/convert_channel_modal';
+import EditChannelHeaderModal from 'components/edit_channel_header_modal';
+import EditChannelPurposeModal from 'components/edit_channel_purpose_modal';
+import * as Menu from 'components/menu';
+import ChannelPermissionGate from 'components/permissions_gates/channel_permission_gate';
+import RenameChannelModal from 'components/rename_channel_modal';
+
+import {Constants, ModalIdentifiers} from 'utils/constants';
+
+type Props = {
+    channel: Channel;
+    isReadonly: boolean;
+    isDefault: boolean;
+}
+
+const ChannelSettingsSubmenu = ({channel, isReadonly, isDefault}: Props): JSX.Element => {
+    const dispatch = useDispatch();
+    const {formatMessage} = useIntl();
+    const channelPropertiesPermission = channel.type === Constants.PRIVATE_CHANNEL ? Permissions.MANAGE_PRIVATE_CHANNEL_PROPERTIES : Permissions.MANAGE_PUBLIC_CHANNEL_PROPERTIES;
+    const handleRenameChannel = () => {
+        dispatch(
+            openModal({
+                modalId: ModalIdentifiers.RENAME_CHANNEL,
+                dialogType: RenameChannelModal,
+                dialogProps: {channel},
+            }),
+        );
+    };
+
+    const handleEditHeader = () => {
+        dispatch(
+            openModal({
+                modalId: ModalIdentifiers.EDIT_CHANNEL_HEADER,
+                dialogType: EditChannelHeaderModal,
+                dialogProps: {channel},
+            }),
+        );
+    };
+
+    const handleEditPurpose = () => {
+        dispatch(
+            openModal({
+                modalId: ModalIdentifiers.EDIT_CHANNEL_PURPOSE,
+                dialogType: EditChannelPurposeModal,
+                dialogProps: {channel},
+            }),
+        );
+    };
+
+    const handleConvertToPrivate = () => {
+        dispatch(
+            openModal({
+                modalId: ModalIdentifiers.CONVERT_CHANNEL,
+                dialogType: ConvertChannelModal,
+                dialogProps: {
+                    channelId: channel.id,
+                    channelDisplayName: channel.display_name,
+                },
+            }),
+        );
+    };
+
+    return (
+        <Menu.SubMenu
+            id={'channelSettings'}
+            labels={
+                <FormattedMessage
+                    id='channelSettings'
+                    defaultMessage='Channel Settings'
+                />
+            }
+            leadingElement={<CogOutlineIcon size={18}/>}
+            trailingElements={<ChevronRightIcon size={16}/>}
+            menuId={'channelSettings-menu'}
+            menuAriaLabel={formatMessage({id: 'channelSettings', defaultMessage: 'Channel Settings'})}
+        >
+            {!isReadonly && (
+                <>
```

Comment:
The change you'd need to make is here. I think we either need to change the fragment to an array and add unique `key` props to each item in the array or split these up so they have their own conditions